### PR TITLE
Add a protected path to network/paths NB API

### LINKED ISSUE #5097: Add "protected" to available paths api
As a user, I can request the availability of protected paths so I'd be able to see how much bandwidth is available to me in case I need a protected service.

API: /v1/network/paths

### PR DESCRIPTION
**Pull request by dmitrii-beliakov**
This feature adds calculation of a protected path for each path returned by `/network/paths` API end point and if the path exists, adds it to the payload under `protected_path` property.
To enable this functionality a user must specify a request param `protected = true`. If this parameter is omitted or false, `network/paths/` behaves as if this feature doesn't exist (nothing new in the payload). 
When a protected path doesn't exist for some path, payload doesn't contain `protected_path` (maybe worth to change it and show and empty object to highlight that the logic has been executed)
Closes #5097

### CODE REVIEW

File: docs/design/solutions/protected-paths/Protected-paths-availability.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+```json
+{
+  "paths": [
+    {
+      "bandwidth": 0,
+      "is_backup_path": true,
+      "latency": 0,
+      "latency_ms": 0,
+      "latency_ns": 0,
+      "is_protected_path_available": true,
```
[the remark this answers is not in the document]
Comment (dmitrii-beliakov) [reply docs/design/solutions/protected-paths/Protected-paths-availability.md:46]: If the `protected` is not true, then the logic of calculating protected availability is not executed at all and the field `is_protected_path_available` is null. With the current setup, a marshler doesn't include null fields into a json, so when a user doesn't explicitly ask for this feature, they get the get-paths-version response. This is really an addition, not a change. It works on top of get paths, just adds more information.
"protected" here is a placeholder I took from the task for a lack of a better name. Probably, it should be "display_protected_availability" or something like that. This is not a filter. This thing enables one more field for the result elements. This way, the result is consistent with the get paths. Additionally, we want to distinguish if there is no path (an empty list) or there is a single path but a protected path cannot be created.
I imagine it like a supermarket: we come and ask "which paths do you have?" and the system tells "here's one path with a lot of bandwidth, but you cannot have it protected, and here's another path with low bandwidth, but buy now and you get a protected path for free".

**Comment by dmitrii-beliakov** [comment]
I want to add more test cases to this feature. But don't know what kind of scenarios to include. Appreciate any input about what to consider.

**Comment by pkazlenka** [comment]
Please improve parameters validation. Request with only src/dst switches and protected = true fails with NPE in org.openkilda.wfm.topology.nbworker.bolts.PathsBolt (`/api/v1/network/paths?src_switch=00:00:00:00:00:00:00:03&dst_switch=00:00:00:00:00:00:00:07&protected=true`

**Comment by pkazlenka** [comment]
Also, regression shows that on this branch creating Y-Flows with protected path is broken. E.g. such a request should create YFlow (at least it does so on develop branch), but causes an error.
```
{
  "y_flow_id": null,
  "shared_endpoint": {
    "switch_id": "00:00:00:00:00:00:00:01",
    "port_number": 10
  },
  "maximum_bandwidth": 1000,
  "path_computation_strategy": "COST",
  "encapsulation_type": "TRANSIT_VLAN",
  "max_latency": null,
  "max_latency_tier2": null,
  "ignore_bandwidth": false,
  "periodic_pings": false,
  "pinned": false,
  "priority": null,
  "strict_bandwidth": false,
  "description": "autotest y-flow: So wise so young, they say, do never live long.",
  "allocate_protected_path": true,
  "diverse_flow_id": null,
  "sub_flows": [
    {
      "flow_id": null,
      "endpoint": {
        "switch_id": "00:00:00:00:00:00:00:09",
        "port_number": 10,
        "vlan_id": 2255,
        "inner_vlan_id": 0
      },
      "shared_endpoint": {
        "vlan_id": 1893,
        "inner_vlan_id": 0
      },
      "description": null
    },
    {
      "flow_id": null,
      "endpoint": {
        "switch_id": "00:00:00:00:00:00:00:08",
        "port_number": 9,
        "vlan_id": 813,
        "inner_vlan_id": 0
      },
      "shared_endpoint": {
        "vlan_id": 3857,
        "inner_vlan_id": 0
      },
      "description": null
    }
  ]
}
```
`ResourcesAllocationAction failed: Not enough bandwidth or no path found. Couldn't find non overlapping protected path`

I can't find anything interesting in app logs (exceptions, sudden failures), so, I assume that the problem is somewhere in path computation algorithm.
Please, fix.

**Comment by dmitrii-beliakov** [comment]
There were some weird issues after merging from develop. I created another branch from develop and applied these changes as a patch, and then adjusted the code to the new parts coming from develop.

This branch and PR is abandoned. The feature is now [here](https://github.com/telstra/open-kilda/pull/5263). 
Regression tests are OK in the new branch.

**Pull request closed without merging**